### PR TITLE
Cql4bonnie bonnie 1187 invalid ucum dosage units

### PR DESCRIPTION
**Pull request by c-monkey**
This PR changes medication units to be ucum compliant, and provides a rake task to fix up existing patients to the new units.

Used 2 references for what these units should be:
1) http://hl7.org/fhir/us/core/2017Jan/ValueSet-us-core-ucum.html
   - {Capsule} for capsule(s)
   - {Dose} for dose(s)

2) http://unitsofmeasure.org/ucum.html#section-Examples-for-some-Non-Units. (note period is in the URL!)
   - g for gram(s)
   - ug for mcg(s)
   - mL for ml(s)
   - {tbl} for tablet(s)

Added 2 rake tasks, intended to be run just once:
- bonnie:patients:fix_non_ucum_dose_and_quantity_dispensed
- bonnie:patients:tabulate_dosage_units

The tabulate task is to count instances, and the fix task does the conversion. Therefore one one could run the tabulate, the fix, and the tabulate a second time to see the changed results.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1187
- [x] JIRA ticket links to this PR
- [X] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [X] Code coverage has not gone down and all code touched or added is covered. 
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [X] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [X] JIRA test links: https://jira.mitre.org/browse/BONNIE-1214 and https://jira.mitre.org/browse/BONNIE-1215
- [X] Justification for using JIRA tests: these tests are to make sure the run-once rake tasks work
- [X] JIRA tests have been added to sprint

**Reviewer 1:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @jbradl11  
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/templates/patient_builder/edit_fulfillments.hbs [hunk 1]
```diff
@@ -20,14 +20,14 @@
       <div class="input-group-addon">
         <label for="quantity_dispensed_unit_{{@cid}}" class="sr-only">quantity dispensed unit</label>
         <select id="quantity_dispensed_unit_{{@cid}}" name="quantity_dispensed_unit" class="form-control medications-control">
-          <option value="capsule(s)">capsule(s)</option>
+          <option value="{Capsule}">capsule(s)</option>
           <option value="dose(s)">dose(s)</option>
```
Comment (jbradl11) [inline app/assets/javascripts/templates/patient_builder/edit_fulfillments.hbs]: Does dose need to be {Dose}?

**Comment by c-monkey** [comment]
As documented in the JIRA ticket, we have 1 patient with an invalid custom code, and there are 4 patients that do not have a facility.code.code - both are revealed by the rake task but not caused by this PR's changes.

**Review by lukoktonos — COMMENTED** [review]
Code looks good.

JIRA tests will pass once https://github.com/projecttacoma/bonnie/pull/913 is merged in.  The only part that isn't passing now is the rake task generating the 2 errors that will be fixed by that merge.

Once merged in, this PR will be approved.

**Comment by c-monkey** [comment]
@losborne @jbradl11 
Using the latest bonnie back bonnie_production-2018-04-24-23-00-04
```
Before bonnie:patients:tabulate_dosage_units:
   {"mg"=>903, "capsule(s)"=>651, "dose(s)"=>3}

bonnie:patients:fix_non_ucum_dose_and_quantity_dispensed
...................................................... Done (5437 records).

After bonnie:patients:fix_non_ucum_dose_and_quantity_dispensed:
   {"mg"=>903, "{Capsule}"=>651, "{Dose}"=>3}
```
No regressions in regression_test script.

https://jira.mitre.org/browse/BONNIE-1215 works fine with the latest database, no need to grab from November. Only issue is can't test QRDA export anymore.

File: lib/tasks/bonnie_run_once.rake [hunk 2]
```diff
@@ -219,7 +219,109 @@ namespace :bonnie do
...
+            puts patient.inspect
+          end
+        end
+      end
+      puts " Done (#{count} records)."
+    end
+
+    desc %{Count each of the existing dosage units
+      $ bundle exec rake bonnie:patients:tabulate_dosage_units}
+    task :tabulate_dosage_units => :environment do
```
Comment (lukoktonos) [inline lib/tasks/bonnie_run_once.rake]: This task tabulates both dose_units and quantity_dispensed_units. I would recommend renaming it to include both (as the task above does already), such as task :tabulate_dose_and_quantity_dispensed_units.

Comment (c-monkey) [reply lib/tasks/bonnie_run_once.rake]: Sounds good. I have also updated the BONNIE-1214 and BONNIE-1215 tests that refer to this rake task. I don't think it's referred to anywhere else.

**Review by lukoktonos — APPROVED** [review]
Looks good. JIRA test passes.

**Review by jbradl11 — APPROVED** [review]
Code looks good. Rake tasks output expected results. Jira Tests pass (besides being able to test QRDA export).

**Pull request merged**